sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -73,15 +79,25 @@ namespace Azure { namespace Core {
      */
     static Uuid CreateUuid()
     {
-      std::random_device rd;
-
       uint8_t uuid[UuidSize] = {};
 
+#if defined(AZ_PLATFORM_WINDOWS)
+      std::random_device rd;
+
       for (int i = 0; i < UuidSize; i += 4)
       {
         const uint32_t x = rd();
         std::memcpy(uuid + i, &x, 4);
       }
+#elif defined(AZ_PLATFORM_POSIX)
+      int ret = RAND_bytes(uuid, UuidSize);
+      if (ret <= 0)
+      {
+        abort();
```

Comment:
Should we wait to merge this until after https://github.com/Azure/azure-sdk-for-cpp/pull/2170 so we can fix this usage?

Also, how do we plan to test this implementation?